primitives/src/fees.rs

```diff
@@ -0,0 +1,119 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate::{constants::MILLI_KILT, Balance};
+use frame_support::{
+	traits::{Currency, Get, Imbalance, OnUnbalanced},
+	weights::{DispatchClass, WeightToFeeCoefficient, WeightToFeeCoefficients, WeightToFeePolynomial},
+};
+use pallet_balances::WeightInfo;
+use smallvec::smallvec;
+use sp_runtime::Perbill;
+
+use crate::{AccountId, NegativeImbalanceOf};
+
+pub struct SplitFeesByRatio<R, Ratio, Beneficiary1, Beneficiary2>(
+	sp_std::marker::PhantomData<(R, Ratio, Beneficiary1, Beneficiary2)>,
+);
+impl<R, Ratio, Beneficiary1, Beneficiary2> OnUnbalanced<NegativeImbalanceOf<R>>
+	for SplitFeesByRatio<R, Ratio, Beneficiary1, Beneficiary2>
+where
+	R: pallet_balances::Config,
+	Beneficiary1: OnUnbalanced<NegativeImbalanceOf<R>>,
+	Beneficiary2: OnUnbalanced<NegativeImbalanceOf<R>>,
+	Ratio: Get<(u32, u32)>,
+{
+	fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalanceOf<R>>) {
+		let ratio = Ratio::get();
+		if let Some(fees) = fees_then_tips.next() {
+			// for fees, 80% to treasury, 20% to author
```

Comment:
```suggest
			// for fees, 50% to treasury, 50% to author
```

Reply:
we don't know the ratio here. So I would delete that line

Reply:
628c62dbd9d670b3cb5217fe0b58424ee5760462